trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/rest/RequestBuilder.java

```diff
@@ -50,19 +54,33 @@ private RequestBuilder() {}
     static Request fromRequest(ContainerRequest request)
     {
         MultiMap requestHeaders = ImmutableMultiMap.copyOfCaseInsensitive(request.getHeaders().entrySet());
-        String xAmzDate = requestHeaders.getFirst("x-amz-date").orElseThrow(() -> {
-            log.debug("Missing \"x-amz-date\" header");
-            return new WebApplicationException(Response.Status.BAD_REQUEST);
-        });
-        Optional<String> securityTokenHeader = requestHeaders.getFirst("x-amz-security-token");
+        SplitQueryParameters queryParameters = splitQueryParameters(ImmutableMultiMap.copyOf(request.getUriInfo().getQueryParameters(true).entrySet()));
+        MultiMap authorizationQueryParameters = ImmutableMultiMap.copyOf(queryParameters.signingQueryParameters());
+        Instant requestTimestamp = AwsTimestamp.fromRequestTimestamp(
+                requestHeaders.getFirst("x-amz-date")
+                    .or(() -> authorizationQueryParameters.getFirst("X-Amz-Date"))
+                    .orElseThrow(() -> {
+                        log.debug("Missing \"x-amz-date\" header");
+                        return new WebApplicationException(Response.Status.BAD_REQUEST);
+                    }));
         RequestContent requestContent = request.hasEntity() ? buildRequestContent(request.getEntityStream(), getRequestContentTypeFromHeader(requestHeaders)) : RequestContent.EMPTY;
+        Optional<String> securityTokenHeader = requestHeaders.getFirst("x-amz-security-token").or(() -> authorizationQueryParameters.getFirst("X-Amz-Security-Token"));
+        RequestAuthorization requestAuthorization = requestHeaders.getFirst("authorization").map(authorizationHeader -> RequestAuthorization.parse(authorizationHeader, securityTokenHeader))
+                .orElseGet(() -> RequestAuthorization.presignedParse(
```

Comment:
It feels like `SigningQueryParameters` could make these headers methods instead of being referenced by string. We end up seeing this raw strings a lot.